roofit/roofitcore/inc/RunContextTracker.h

```diff
@@ -0,0 +1,43 @@
+// Author: Jonas Rembser, CERN  Feb 2021
+
+/*****************************************************************************
+ * RooFit
+ * Authors:                                                                  *
+ *   WV, Wouter Verkerke, UC Santa Barbara, verkerke@slac.stanford.edu       *
+ *   DK, David Kirkby,    UC Irvine,         dkirkby@uci.edu                 *
+ *                                                                           *
+ * Copyright (c) 2000-2020, Regents of the University of California          *
+ *                          and Stanford University. All rights reserved.    *
+ *                                                                           *
+ * Redistribution and use in source and binary forms,                        *
+ * with or without modification, are permitted according to the terms        *
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)             *
+ *****************************************************************************/
+
+#ifndef roofit_roofitcore_RunContextTracker_h
+#define roofit_roofitcore_RunContextTracker_h
+
+#include "RooChangeTracker.h"
+#include "RunContext.h"
+
+class RunContextTracker {
+
+   // Enable logging by adding this line:
+   //     RooMsgService::instance().addStream(DEBUG, Topic(FastEvaluations));
+
+public:
+   RunContextTracker(RooBatchCompute::RunContext const &runContext);
+
+   void resetTrackers();
+
+   void cleanRunContext(RooAbsArg const &caller, RooBatchCompute::RunContext &runContext);
+
+private:
+   void addTracker(const RooAbsReal *absReal);
+
+   // member variables
+
+   std::unordered_map<const RooAbsReal *, std::unique_ptr<RooChangeTracker>> trackers_;
```

Comment:
RooFit convention would be
```suggestion
   std::unordered_map<const RooAbsReal *, std::unique_ptr<RooChangeTracker>> _trackers;
```

I suggest to not start a third convention inside root ...

Reply:
Sure!